README.md

```diff
@@ -36,14 +36,39 @@ mockServer.AddVerification(
     ))
 ```
 
-Expectation Defaults:
+Create a verification sequence example:
+```
+mockServer := mockclient.Client{
+    T: t,
+    BaseURL: os.Getenv("MOCKSERVER_HOST"),
+}
+
+mockServer.AddVerificationSequence(
+    mockclient.CreateVerification(
+        mockclient.WhenRequestPath("/a"),
+    ),
+    mockclient.CreateVerification(
```

Comment:
I feel this is a little unintuitive as an API since the "CreateVerification" is in the "regular" use-case independent of the other "CreateVerification" calls. So it's a little weird here that they do depend on each other here. I would almost say that the whole sequence itself should be a single "CreateVerification" call. I was thinking something more like:
```
mockServer.AddVerificationSequence(
	mockclient.CreateVerification(
	mockclient.WhenCallSequence(
		{
			"/path1",
			http.MethodPOST,
		},
		{
			"/path2"
		}
		),
	),
)
```
But I don't have any good suggestions myself that don't break/modify the base stuff so this is good enough for now to get this functionality.

Reply:
Long answer as I struggled with this for a while :) Isn't the "closure" the `AddVerification` & `AddVerificationSequence` call? So everything inside that function call is what defines that verification. Doesn't that hold true for both `AddVerification` and `AddVerificationSequence`? The `CreateVerification` function is just creating the "requests" that will be verified. So the semantic is a bit different for the 2 use-cases. In `AddVerification` we are only verifying one thing (sum of requests) but in `AddVerificationSequence` we are verifying many things (specific requests).  In summary, to reuse the API we had already created and not have something quite alien this was the best I could come up with.

The API suggestion above is using JSON which are API is supposed to wrap, we have a `RequestMatcher` object for exactly this.

Reply:
Hmm, I guess that makes sense. I think what I am mostly struggling with is that in AddVerification the order of the Verifications created doesn't matter, but in AddVerificationSequence it does, but perhaps that is obvious enough from the name?

Reply:
I'll add a clarification to the README and comments be make that clear.